src/s3fs.cpp

```diff
@@ -644,6 +644,33 @@ static int check_parent_object_access(const char* path, int mask)
     return 0;
 }
 
+//
+// Check if parent allows subdirectories
+//
+static int check_parent_allows_subdirectory(const char* path, int mask)
+{
+    std::string parent;
+    int result;
+    headers_t meta;
+
+    S3FS_PRN_DBG("[path=%s]", path);
+
+    parent = mydirname(path);
+    if(parent == "."){
+        parent = "/";
```

Comment:
Why is the parent set to the root in this case?

Reply:
Good q, yeah to be completely honest I found this a bit odd too, I was mainly following the existing pattern: https://github.com/wayscript/s3fs-fuse/blob/jack/ws-restrictions/src/s3fs.cpp#L636

Hadnt investigated it yet, but my current guess was that s3fs-fuse always gets handed the full path relative to the mount point, and so if dirname of that path returned '.' that's functionally the same as saying the dir is the mount point directory.

Reply:
Weird ok